db/import_column_family_job.cc

```diff
@@ -29,79 +29,132 @@ namespace ROCKSDB_NAMESPACE {
+      if (!status.ok()) {
+        return status;
+      }
 
-  // Read the information of files we are importing
-  for (const auto& file_metadata : metadata_) {
-    const auto file_path = file_metadata.db_path + "/" + file_metadata.name;
-    IngestedFileInfo file_to_import;
-    status = GetIngestedFileInfo(file_path, next_file_number++, sv,
-                                 file_metadata, &file_to_import);
-    if (!status.ok()) {
-      return status;
-    }
-    files_to_import_.push_back(file_to_import);
-  }
+      if (file_to_import.num_entries == 0) {
+        status = Status::InvalidArgument("File contain no entries");
+        return status;
+      }
 
-  auto num_files = files_to_import_.size();
-  if (num_files == 0) {
-    status = Status::InvalidArgument("The list of files is empty");
-    return status;
-  }
+      if (!file_to_import.smallest_internal_key.Valid() ||
+          !file_to_import.largest_internal_key.Valid()) {
+        status = Status::Corruption("File has corrupted keys");
+        return status;
+      }
+
+      files_to_import_per_cf.push_back(file_to_import);
+      num_files++;
+
+      // Calculate the smallest and largest keys of all files in this CF
+      if (i == 0) {
+        smallest = file_to_import.smallest_internal_key;
+        largest = file_to_import.largest_internal_key;
+      } else {
+        if (cfd_->internal_comparator().Compare(
+                smallest, file_to_import.smallest_internal_key) < 0) {
+          smallest = file_to_import.smallest_internal_key;
+        }
+        if (cfd_->internal_comparator().Compare(
+                largest, file_to_import.largest_internal_key) > 0) {
+          largest = file_to_import.largest_internal_key;
+        }
+      }
+    }
 
-  for (const auto& f : files_to_import_) {
-    if (f.num_entries == 0) {
-      status = Status::InvalidArgument("File contain no entries");
+    if (num_files == 0) {
+      status = Status::InvalidArgument("The list of files is empty");
       return status;
     }
+    files_to_import_.push_back(files_to_import_per_cf);
+    cf_file_info.smallest_internal_key = smallest;
+    cf_file_info.largest_internal_key = largest;
+    cf_ingest_infos.push_back(cf_file_info);
+  }
 
-    if (!f.smallest_internal_key.Valid() || !f.largest_internal_key.Valid()) {
-      status = Status::Corruption("File has corrupted keys");
+  std::sort(cf_ingest_infos.begin(), cf_ingest_infos.end(),
+            [this](const ColumnFamilyIngestFileInfo& info1,
+                   const ColumnFamilyIngestFileInfo& info2) {
+              return cfd_->internal_comparator().Compare(
+                         info1.smallest_internal_key,
+                         info2.smallest_internal_key) < 0;
+            });
+
+  for (size_t i = 0; i + 1 < cf_ingest_infos.size(); i++) {
+    if (cfd_->internal_comparator().Compare(
+            cf_ingest_infos[i].largest_internal_key,
+            cf_ingest_infos[i + 1].smallest_internal_key) >= 0) {
```

Comment:
Should we just compare user key here and require that CFs should not overlap in user key range for the API?